server.js

```diff
@@ -59,11 +63,15 @@ app.delete('/items/:id', function(req, res) {
   }
 });
 
+// Route that updates item name based on specified id
 app.put('/items/:id', jsonParser, function(req, res) {
   var id = parseInt(req.params.id);
   var body = req.body;
-  var itemName; // the new name
+  var itemName;
   var idFound = false;
+  if (id != req.body.id) {
+    return res.sendStatus(400);
+  }
   if (req.body.hasOwnProperty("name")) {
```

Comment:
If a `put` request is sent without a `name` field, I think this handler will not send back a response. Might need to do a check for the existence of the `name` field so a `400` response code can be sent back.